airflow/contrib/hooks/bigquery_hook.py

```diff
@@ -934,13 +934,22 @@ def var_print(var_name):
         else:
             return "Format exception for {var}: ".format(var=var_name)
 
-    cmpt = table_input.split(':')
+    if table_input.count('.') + table_input.count(':') > 3:
+        raise Exception((
+            '{var}Use either : or . to specify project '
+            'got {input}'
+        ).format(var=var_print(var_name), input=table_input))
+
+    cmpt = table_input.rsplit(':', 1)
+    project_id = None
+    rest = table_input
     if len(cmpt) == 1:
         project_id = None
         rest = cmpt[0]
-    elif len(cmpt) == 2:
-        project_id = cmpt[0]
-        rest = cmpt[1]
+    elif len(cmpt) == 2 and cmpt[0].count(':') <= 1:
```

Comment:
Same answer here, this is not a valid project_id, and it will be caught by L937.